### DISCUSSION TITLE
Using sybilSBML on already-generated .RDS files

### DISCUSSION THREAD

**Discussion by arbujold** [jotech/gapseq#120] [Q&A]
When I first ran all of my data through gapseq using the doall command, I hadn't been able to get sybilSBML to load properly, so no .XML files were created. I have since been able to get sybilSBML to load and would like to general the .XML files from the .RDS file outputs I already have. 
Is it possible to do this? If so, how?

**Comment by Waschina** [comment]
Good question. 

What you could try is to adjust and run the following commands in R:

```R
library(sybil)
library(sybilSBML)
source("<path_to_gapseq>/src/sbml_write.R")

mod <- readRDS("<your_model>.RDS")
write_gapseq_sbml(mod, "<your_model>.xml")
```

please make sure to adjust the `<...>`-Parts to your filenames and paths.

Let me know if that works.

Best
Silvio

**Reply by arbujold** [reply]
Hi, I have been having some troubles making this work. The latest issue gives this message:
> write_gapseq_sbml(mod, "FILE.xml")
Could not write file FILE.xml
Patching file FILE.xml ...Error in (grp_block_start + 1):n_lines : argument of length 0
In addition: Warning messages:
1: In write_gapseq_sbml(mod, "FILE") : Writing SBML-file failed.
2: In readLines(paste0(out.id, ".xml")) :
  incomplete final line found on 'FILE.xml'

**Reply by Waschina** [reply]
could you try
```R
library(sybilSBML)
mod <- readRDS("<your_model>.RDS")
sybilSBML::writeSBML(mod, filename = "<your_model>.RDS",
                     level = 3, version = 1, fbcLevel = 2, 
                     printNotes = T, printAnnos = T)
```

If that works, the final xml might have some issues, but we can be sure that your sybilSBML and libSBML installation work as expected.
